### ISSUE TITLE
unable to run testsuite from src rpm package

### ISSUE THREAD

**Issue by jscotka** [cockpit-project/cockpit#4494]
Hi,
I tried to run upstream testsuite from src rpm package for Red Hat Enterprise Linux 7,
first issue what I've met were: 
```
$ ./verify/testsuite-prepare 
/home/test/cockpit/test/../tools/make-source: line 39: ../autogen.sh: No such file or directory
Command '['/home/test/cockpit/test/../tools/make-source']' returned non-zero exit status 1
```
this trouble I've improved by: 
```
touch ../.tarball
```
but after that touching, another issue: 
```
./verify/testsuite-prepare 
checking for a BSD-compatible install... /bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for ar... ar
checking the archiver (ar) interface... ar
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for gcc option to accept ISO C99... -std=gnu99
checking for ranlib... ranlib
checking for library containing strerror... none required
checking for ANSI C header files... (cached) yes
checking whether ln -s works... yes
checking whether make supports nested variables... (cached) yes
checking whether to install to prefix only... no
checking for pkg-config... /bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for GIO... yes
checking for SYSTEMD... yes
checking for JSON_GLIB... yes
checking for POLKIT... yes
checking for krb5-config... /bin/krb5-config
checking for working krb5-config... yes
checking build with cockpit-ws... yes
checking for GLIB_TLS... yes
checking for LIBSSH... yes
checking security/pam_appl.h usability... yes
checking security/pam_appl.h presence... yes
checking for security/pam_appl.h... yes
checking for pam_open_session in -lpam... yes
checking keyutils.h usability... yes
checking keyutils.h presence... yes
checking for keyutils.h... yes
checking for keyctl_join_session_keyring in -lkeyutils... yes
checking crypt.h usability... yes
checking crypt.h presence... yes
checking for crypt.h... yes
checking for crypt_r in -lcrypt... yes
checking whether to build with PCP... yes
checking pcp/pmapi.h usability... yes
checking pcp/pmapi.h presence... yes
checking for pcp/pmapi.h... yes
checking for pcp/import.h... yes
checking for pcp/pmda.h... yes
checking for pmNewContext in -lpcp... yes
checking for pmdaCacheLookup in -lpcp_pmda... yes
checking for pmiStart in -lpcp_import... yes
checking for gss_import_cred in -lgssapi_krb5... yes
checking for ssh_gssapi_set_creds in -lssh... yes
checking for ssh_set_agent_socket in -lssh... yes
checking for systemd unit dir... /usr/lib/systemd/system
checking whether NLS is requested... yes
checking for intltool-update... /bin/intltool-update
checking for intltool-merge... /bin/intltool-merge
checking for intltool-extract... /bin/intltool-extract
checking for xgettext... /bin/xgettext
checking for msgmerge... /bin/msgmerge
checking for msgfmt... /bin/msgfmt
checking for gmsgfmt... /bin/msgfmt
checking for perl... /bin/perl
checking for perl >= 5.8.1... 5.16.3
checking for XML::Parser... ok
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking for LC_MESSAGES... yes
checking libintl.h usability... yes
checking libintl.h presence... yes
checking for libintl.h... yes
checking for ngettext in libc... yes
checking for dgettext in libc... yes
checking for bind_textdomain_codeset... yes
checking for msgfmt... (cached) /bin/msgfmt
checking for dcgettext... yes
checking if msgfmt accepts -c... yes
checking for gmsgfmt... (cached) /bin/msgfmt
checking for xgettext... (cached) /bin/xgettext
checking for phantomjs... /bin/phantomjs
checking for go... /bin/go
checking for usermod... /usr/sbin/usermod
checking for newusers... /usr/sbin/newusers
checking whether /usr/sbin/newusers understands --crypt-method... yes
checking for chpasswd... /usr/sbin/chpasswd
checking for pkexec... /bin/pkexec
checking for sudo... /bin/sudo
checking for ssh-add... /bin/ssh-add
checking for ssh-agent... /bin/ssh-agent
checking for chcon... /bin/chcon
checking for fdwalk... no
checking for asan flags... no
checking whether to build documentation... yes
checking for xsltproc... /bin/xsltproc
checking for xmlto... /bin/xmlto
checking for branding directory... default
checking for debug mode... default
checking whether to build with coverage... no
checking build strict... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doc/guide/version
config.status: creating po/Makefile.in
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing default-1 commands
config.status: executing po/stamp-it commands

               Cockpit 0.107
             ================

        prefix:                     /usr/local
        exec_prefix:                ${prefix}
        libdir:                     ${exec_prefix}/lib
        libexecdir:                 ${exec_prefix}/libexec
        bindir:                     ${exec_prefix}/bin
        sbindir:                    ${exec_prefix}/sbin
        datarootdir:                ${prefix}/share
        datadir:                    ${datarootdir}
        sysconfdir:                 ${prefix}/etc
        localstatedir:              ${prefix}/var
        pamdir:                     ${libdir}/security

        compiler:                   gcc -std=gnu99
        cflags:                     -Wall           -Werror=strict-prototypes -Werror=missing-prototypes           -Werror=implicit-function-declaration           -Werror=pointer-arith -Werror=init-self           -Werror=format-security           -Werror=missing-include-dirs -g -O2 -g
        cppflags:                   

        cockpit-ws:                 yes
        cockpit-ws user:            root
        cockpit-ws group:           
        selinux config type:        

        Maintainer mode:            yes
        Building docs:              yes
        Debug mode:                 default
        With coverage:              no
        With address sanitizer:     no
        With PCP:                   yes
	Branding:                   default
        Supports key auth:          yes

        pkexec:                     /bin/pkexec
        ssh-add:                    /bin/ssh-add
        ssh-agent:                  /bin/ssh-agent
        sudo:                       /bin/sudo
        usermod:                    /usr/sbin/usermod
        chcon:                      /bin/chcon

/home/test/cockpit/tools/missing: line 81: aclocal-1.15: command not found
WARNING: 'aclocal-1.15' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
make: *** [../aclocal.m4] Error 127
Command '['/home/test/cockpit/test/../tools/make-source']' returned non-zero exit status 2
```

there is package `automake-1.13.4-3.el7.noarch` not higher version.
It is strange, because recompiling via rpmbuild --rebuild cockpit...src.rpm works well, so that

**Comment by dperpeet** [comment]
@jscotka can you verify this happened with the current tarball? I believe @larsu fixes this issue already

**Comment by larsu** [comment]
Yeah, 989ced6 should have fixed it. It was originally reported as #4398.

**Comment by stefwalter** [comment]
@jscotka Why isn't .tarball in the top level directory? It's in the tarball.

**Comment by jscotka** [comment]
I've used @dperpeet scratch buid for rhel7 so it shoul be there?. It is questin why it is not in top dir. I unpacked source code via options in rpmbuil (because i need also to have spurces fully patched. Coul it be removed by specfile?

**Comment by dperpeet** [comment]
The spec file doesn't remove anything like that.

**Comment by larsu** [comment]
I think I've found the issue: we didn't include all files when running `make dist` from a tarball and not a git checkout. Please let me know if my fix in #4507 works.
